libcontainer/utils/utils_unix.go

```diff
@@ -3,33 +3,54 @@
 package utils
 
 import (
-	"io/ioutil"
+	"fmt"
 	"os"
 	"strconv"
 
 	"golang.org/x/sys/unix"
 )
 
+// IsProcHandle returns whether or not the given file handle is on procfs.
+func IsProcHandle(fh *os.File) (bool, error) {
+	var buf unix.Statfs_t
+	err := unix.Fstatfs(int(fh.Fd()), &buf)
+	return buf.Type == unix.PROC_SUPER_MAGIC, err
```

Comment:
nit: `IsProcHandle` returns only bool is simpler, as caller has no way to handle error sanely so it should just abort with `/proc is not what we expected`

Reply:
I'll switch it to only return `error` (so at least you know why it failed).